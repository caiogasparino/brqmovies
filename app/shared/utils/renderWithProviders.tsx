import { render, RenderAPI } from "@testing-library/react-native";
import React from "react";
import { SafeAreaProvider } from "react-native-safe-area-context";

import { NavigationContainer } from "@react-navigation/native";
import { THEME_LIGHT } from "app/shared/design/themes/theme-light";
import { ThemeProvider } from "styled-components";

const renderWithProviders = (component: React.ReactElement): RenderAPI => {
    return render(
        <NavigationContainer>
            <SafeAreaProvider>
                <ThemeProvider theme={THEME_LIGHT}>{component}</ThemeProvider>
            </SafeAreaProvider>
        </NavigationContainer>,
    );
};

export { renderWithProviders };
