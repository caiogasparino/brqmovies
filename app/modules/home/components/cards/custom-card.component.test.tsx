import { renderWithProviders } from "app/shared/utils/renderWithProviders";

import { CustomCard } from "./custom-card.component";

describe("CustomCard component", () => {
    it("matches snapshot", () => {
        const { toJSON } = renderWithProviders(
            <CustomCard
                icon="home"
                label1="Test Label 1"
                label2="Test Label 2"
                onPress={() => {}}
            />,
        );
        expect(toJSON()).toMatchSnapshot();
    });
});
