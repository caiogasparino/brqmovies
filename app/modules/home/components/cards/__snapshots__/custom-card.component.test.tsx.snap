// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard component matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
